deployment/ansible/deploy.yml

```diff
@@ -17,103 +17,138 @@
-      file:
-        path: "{{ project_root }}/{{ config.keypair.dir }}"
-        state: directory
-        mode: 0700
-      tags: [config]
-
-    - name: "Copy node keys to {{ project_root }}/{{ config.keypair.dir }}"
-      copy:
-        src: "{{ item }}"
-        dest: "{{ project_root }}/{{ config.keypair.dir }}"
-        mode: 0600
-      with_items: "{{ keypair }}"
-      when: keypair is defined
-      tags: [config]
-
-    - name: Update epoch configuration
-      template:
-        src: templates/epoch.yaml.j2
-        dest: "{{ project_root }}/epoch.yaml"
-        mode: '0600'
-      tags: [config]
-
-    - name: Update accounts configuration
-      copy:
-        content: "{{ genesis_accounts | to_nice_json }}"
-        dest: "{{ genesis_accounts_path }}"
-      when: genesis_accounts is defined
-      tags: [config]
-
-    - name: Start epoch daemon
-      command: "{{ project_root }}/bin/epoch start"
-      tags: [daemon]
+        tags: [config]
+
+      - name: "Make sure keys dir ( {{ config.keypair.dir }} ) exists"
+        file:
+          path: "{{ project_root }}/{{ config.keypair.dir }}"
+          state: directory
+          mode: 0700
+        tags: [config]
+
+      - name: "Copy node keys to {{ project_root }}/{{ config.keypair.dir }}"
+        copy:
+          src: "{{ item }}"
+          dest: "{{ project_root }}/{{ config.keypair.dir }}"
+          mode: 0600
+        with_items: "{{ keypair }}"
+        when: keypair is defined
+        tags: [config]
+
+      - name: Update epoch configuration
+        template:
+          src: templates/epoch.yaml.j2
+          dest: "{{ project_root }}/epoch.yaml"
+          mode: '0600'
+        tags: [config]
+
+      - name: Update accounts configuration
+        copy:
+          content: "{{ genesis_accounts | to_nice_json }}"
+          dest: "{{ genesis_accounts_path }}"
+        when: genesis_accounts is defined
+        tags: [config]
+
+      - name: Start epoch daemon
+        command: "{{ project_root }}/bin/epoch start"
+        tags: [daemon]
+
+      - name: Read app version
+        command: "cat {{ project_root }}/VERSION"
+        register: version_output
+        tags: [datadog-event]
+
+      - name: Read app revision
+        command: "cat {{ project_root }}/REVISION"
+        register: revision_output
+        tags: [datadog-event]
+
+      - name: Send Datadog event
+        datadog_event:
+          alert_type: success
+          title: Deploy
+          text: |
+            %%%
+            Revision: [{{ revision_output.stdout }}](https://github.com/aeternity/epoch/commit/{{ revision_output.stdout }})
+            Package: {{ remote_package | default(local_package) | basename }}
+            %%%
+          api_key: "{{ datadog_api_key }}"
+          app_key: "{{ datadog_app_key }}"
+          tags: ["env:{{ env }}", "version:{{ version_output.stdout }}"]
+        tags: [datadog-event]
+      rescue:
```

Comment:
Does rescuing keep the error, or does it convert it to success?

Reply:
It keeps the error, thus the ansible-playbook still should exit with status 1.
Example (some IP/paths masked):

```
---snip

TASK [API health check] *************************************************************************************************************************************************************************************
Tuesday 09 January 2018  12:02:14 +0200 (0:00:00.706)       0:00:23.263 ******* 
fatal: [ae-dev1-epoch-n0]: FAILED! => {"changed": false, "content": "", "failed": true, "msg": "Status code was not [200]: Request failed: <urlopen error [Errno 61] Connection refused>", "redirected": false, "status": -1, "url": "http://x.x.x.x:3013/v1/top"}

NO MORE HOSTS LEFT ******************************************************************************************************************************************************************************************

NO MORE HOSTS LEFT ******************************************************************************************************************************************************************************************
	to retry, use: --limit @xxxxxxx/epoch/deployment/ansible/deploy.retry

PLAY RECAP **************************************************************************************************************************************************************************************************
ae-dev1-epoch-n0           : ok=12   changed=4    unreachable=0    failed=1  

---snip
```

Reply:
Good catch, it turns out the rescue also swallows the error and ansible-playbook exit with 0

Reply:
This should fix that issue fa854de